src/includes/traits/Plugin/UpdateUtils.php

```diff
@@ -128,12 +128,18 @@ public function preSiteTransientUpdatePlugins($transient)
         if (empty($_r['_wpnonce']) || !wp_verify_nonce((string) $_r['_wpnonce'], 'upgrade-plugin_'.plugin_basename(PLUGIN_FILE))) {
             return $transient; // Nothing to do here.
         }
-        if (empty($_r[GLOBAL_NS.'_update_pro_version']) || empty($_r[GLOBAL_NS.'_update_pro_zip'])) {
+        if (empty($_r[GLOBAL_NS.'_update_pro_version'])) {
             return $transient; // Nothing to do here.
         }
+
         $update_pro_version = (string) $_r[GLOBAL_NS.'_update_pro_version'];
-        $update_pro_zip     = base64_decode((string) $_r[GLOBAL_NS.'_update_pro_zip'], true);
-        // @TODO Encrypt/decrypt to avoid mod_security issues. Base64 is not enough.
+
+        if (false === ($update_pro_zip = get_site_transient(GLOBAL_NS.'_update_pro_zip_'.$update_pro_version))) {
```

Comment:
You probably want to fail here if `$update_pro_zip` is empty in any way, not just if it is is `false` explicitly.

```php
if(!($update_pro_zip = get_site_transient(GLOBAL_NS.'_update_pro_zip_'.$update_pro_version))) {
    return $transient;
}
```